src/java/com/unifina/data/Event.java

```diff
@@ -0,0 +1,52 @@
+package com.unifina.data;
+
+import java.util.Date;
+import java.util.function.Consumer;
+
+public class Event<ContentClass> implements Comparable<Event<ContentClass>> {
```

Comment:
What's the definition of ContentClass? It would be easier to read something like:
`public class Event<ContentClass extends SomeOtherClass>`

Reply:
`Event`s can carry arbitrary objects as payloads. I could write

`public class Event<ContentClass extends Object>`

but IMO it would not make the code easier to read.

However I added comments to explain the purpose of `ContentClass` generic, which is to ensure the `content` and `consumer` are compatible.